### PR TITLE
Refactor Command.completion

### PR DESCRIPTION
**Pull request by rcorre**

### CODE REVIEW

File: qutebrowser/completion/completer.py [hunk 1]
```diff
@@ -204,25 +204,18 @@ def _get_new_completion(self, parts, cursor_part):
...
-            # More arguments than completions
-            log.completion.debug("completions: {}".format(
-                ', '.join(dbg_completions)))
+            # user provided more positional arguments than the command takes
+            return None
+        if completion is None:
             return None
-        dbg_completions[idx] = '*' + dbg_completions[idx] + '*'
-        log.completion.debug("completions: {}".format(
-            ', '.join(dbg_completions)))
```
Comment (The-Compiler) [inline qutebrowser/completion/completer.py]: Why did you remove the debug output here?

File: tests/unit/completion/test_completer.py [hunk 2]
```diff
@@ -143,21 +168,17 @@ def _validate_cmd_prompt(cmd, txt):
...
     (':bind <c-x> |', usertypes.Completion.command),
     (':bind <c-x> foo|', usertypes.Completion.command),
-    (':bind <c-x>| foo', usertypes.Completion.empty),
+    (':bind <c-x>| foo', None),
     (':set| general ', usertypes.Completion.command),
     (':|set general ', usertypes.Completion.command),
     (':set gene|ral ignore-case', usertypes.Completion.section),
     (':|', usertypes.Completion.command),
     (':   |', usertypes.Completion.command),
-    (':bookmark-load      |', usertypes.Completion.bookmark_by_url),
```
Comment (The-Compiler) [inline tests/unit/completion/test_completer.py]: Why did you remove some of the tests here?

File: qutebrowser/completion/completer.py (hunk 1 above)
Comment (rcorre) [inline qutebrowser/completion/completer.py]: those particular traces were no longer relevant, but I could add some relevant ones

File: tests/unit/completion/test_completer.py (hunk 2 above)
Comment (rcorre) [inline tests/unit/completion/test_completer.py]: They were redundant with some of the other test cases e.g. the fake
`bookmark-load` was testing pretty much the same thing as the fake
`quickmark-load` -- a command with a single positional argument with
completions.

Comment (The-Compiler) [inline tests/unit/completion/test_completer.py]: Okay, makes sense :smile:

File: qutebrowser/completion/completer.py (hunk 1 above)
Comment (The-Compiler) [inline qutebrowser/completion/completer.py]: I'm fine with leaving them out and adding new ones when we feel we actually need them.

**Comment by The-Compiler** [comment]
Thanks!
